src/platforms/linux/linuxcryptosettings.cpp

```diff
@@ -46,108 +55,54 @@ LinuxCryptoSettings::LinuxCryptoSettings() {
+  logger.debug() << "Removing legacy keys from the keychain";
+  m_legacyKey.clear();
 
   GError* error = nullptr;
-  gboolean ok = secret_password_clear_sync(cryptosettings_get_schema(), nullptr,
-                                           &error, nullptr);
-  if (!ok) {
-    Q_ASSERT(error);
+  auto schema = cryptosettings_get_schema();
+  secret_password_clear_sync(schema, nullptr, &error, nullptr);
+  if (error != nullptr) {
     logger.error() << "Key reset failed:" << error->message;
     g_error_free(error);
-
-    // Fallback to unencrypted settings.
-    m_keyVersion = CryptoSettings::NoEncryption;
   }
-
-  m_key.clear();
 }
 
-QByteArray LinuxCryptoSettings::getKey(const QByteArray& metadata) {
-  Q_UNUSED(metadata);
-
-  if (m_keyVersion == CryptoSettings::NoEncryption) {
-    logger.error() << "libsecrets is not supported";
-    return QByteArray();
-  }
-
-  if (m_key.isEmpty()) {
-    // Try to lookup the encryption key.
-    GError* error = nullptr;
-    gchar* password = secret_password_lookup_sync(cryptosettings_get_schema(),
-                                                  nullptr, &error, nullptr);
-    if (error != nullptr) {
-      logger.error() << "Key lookup failed:" << error->message;
-      g_error_free(error);
-      error = nullptr;
-      // fall-through to try creating the password anyways
-    }
-
-    if (password != nullptr) {
-      QString b64key(password);
-      m_key = QByteArray::fromBase64(b64key.toUtf8());
-      secret_password_free(password);
-    } else {
-      logger.debug() << "Key not found. Let's create it.";
-      m_key = generateRandomBytes(CRYPTO_SETTINGS_KEY_SIZE);
-
-      QString b64key(m_key.toBase64());
-      gboolean ok = secret_password_store_sync(
-          cryptosettings_get_schema(), SECRET_COLLECTION_DEFAULT,
-          Constants::LINUX_CRYPTO_SETTINGS_DESC, qPrintable(b64key), nullptr,
-          &error, nullptr);
-      if (!ok) {
-        Q_ASSERT(error);
-        logger.error() << "Key storage failed:" << error->message;
-        g_error_free(error);
-
-        // Fallback to unencrypted settings.
-        m_keyVersion = CryptoSettings::NoEncryption;
-        m_key.clear();
-        return QByteArray();
-      }
-    }
-  }
-
-  return m_key;
+void LinuxCryptoSettings::resetKey() {
+  clearLegacyKey();
+  XdgCryptoSettings::resetKey();
 }
 
-CryptoSettings::Version LinuxCryptoSettings::getSupportedVersion() {
-  return m_keyVersion;
+QByteArray LinuxCryptoSettings::getKey(Version version,
+                                       const QByteArray& metadata) {
+  if (version == EncryptionChachaPolyV1) {
+    // A legacy key is being requested - get it from the libsecrets API.
+    return m_legacyKey;
+  } else {
+    // Otherwise, use the XDG secrets portal for all new files and keys.
+    // Clear the legacy key once an attempt has been made to write a new file.
+    clearLegacyKey();
+    return XdgCryptoSettings::getKey(version, metadata);
```

Comment:
Should we confirm this returns a key? And if it returns an empty bytearray try to fall back to legacy? Not sure what scenarios, if any, this would help in.